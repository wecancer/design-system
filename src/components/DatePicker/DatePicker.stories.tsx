import React, {useState} from 'react'
import {Story} from '@storybook/react'
import DatePicker, {Props, DatePickerValue} from '.'
import DatePickerRange, {Props as RangeProps, RangePickerValue} from './Range'

const Template: Story<Props> = () => {
  const [date, setValue] = useState<DatePickerValue | undefined>(new Date(Date.now()))
  return <DatePicker value={date} onChange={({value}) => setValue(value)} />
}
export const Default = Template.bind({})
Default.args = {
  fromYear: 2018,
  toYear: 2025,
}

const RangeTemplate: Story<RangeProps> = (args) => {
  const [date, setValue] = useState<RangePickerValue>({from: new Date(2020, 10, 10), to: new Date(2020, 11, 18)})
  return <DatePickerRange {...args} value={date} onChange={({value}) => setValue(value)} />
}
export const Range = RangeTemplate.bind({})
Range.args = {
  fromYear: 2018,
  toYear: 2025,
}

export default {
  title: 'Components/DatePicker',
  component: DatePicker,
}
