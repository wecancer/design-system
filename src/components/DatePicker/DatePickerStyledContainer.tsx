import styled, {css} from 'styled-components'

export default styled.div`
  ${({theme}) => css`
    position: absolute;
    top: calc(100% + 10px);
    left: 0;
    border-radius: 1rem;
    background-color: ${theme.colors.inputBg};

    .rdp {
      &.rdp-multiple_months .rdp-caption_start .rdp-caption {
        & > h2 {
          order: 2;
        }

        & > .rdp-nav {
          order: 1;
        }
      }

      .rdp-caption_dropdowns {
        display: flex;
        justify-content: center;
        flex: 1;
      }

      .rdp-months {
        display: flex;

        .rdp-month {
          padding: 18px 24px;
        }

        .rdp-month:nth-child(2) {
          border-left: 1px solid rgba(0, 0, 0, 0.1);
        }
      }

      .rdp-dropdown_month {
        position: relative;

        &::after {
          content: '';
          position: absolute;
          bottom: 6px;
          right: 6px;

          width: 0;
          height: 0;
          border-style: solid;
          border-width: 0 0 0.5rem 0.5rem;
          border-color: transparent transparent ${theme.colors.primary} transparent;
          box-shadow: 0px 1px 1px rgba(0, 14, 51, 0.05);
        }

        .rdp-caption_label {
          display: none;
        }

        .rdp-dropdown {
          font-size: 1.0625rem;
          font-weight: ${theme.font.weights.semiBold};
          border: none;
          appearance: none;
          padding: 0.625rem 0.75rem;
          border-radius: 0.375rem;
          text-align: center;
          background-color: ${theme.colors.offWhite};

          &:not(:last-child) {
            margin-right: 0.25rem;
          }
        }
      }

      .rdp-caption {
        display: flex;
        justify-content: space-between;
        align-items: center;
        margin-bottom: 8px;

        .rdp-caption_label {
          font-weight: normal;
          font-size: 1rem;
        }

        .rdp-nav {
          display: flex;
        }

        .rdp-nav_button {
          width: 34px;
          height: 34px;
          appearance: none;
          padding: none;
          border: none;
          cursor: pointer;
          background-color: transparent;
          transition: all ease 250ms;
          display: flex;
          align-items: center;
          justify-content: center;
          border-radius: 50%;

          svg {
            width: 12px;
            height: 12px;
          }

          &:hover {
            background-color: rgba(255, 255, 255, 0.5);
          }
        }
      }

      .rdp-vhidden {
        display: none;
      }

      .rdp-head_cell {
        font-size: 13px;
        text-transform: lowercase;
        color: ${theme.colors.label};
        font-weight: ${theme.font.weights.semiBold};
        padding-bottom: 8px;
      }

      .rdp-row .rdp-cell:not(:first-child) .rdp-day_range_end::before,
      .rdp-row .rdp-cell:not(:first-child) .rdp-day_range_middle::after {
        content: '';
        width: 4px;
        height: 34px;
        position: absolute;
        top: 1px;
        left: -3px;
        background-color: ${theme.colors.bgPrimary};
      }

      .rdp-cell {
        width: 38px;
        height: 34px;
        padding: 1px;
        position: relative;

        button {
          font-weight: ${theme.font.weights.semiBold};
          font-size: 14px;
          width: 100%;
          height: 100%;
          padding: 0;
          border-radius: 0.375rem;
          border: none;
          appearance: none;
          background-color: transparent;
          cursor: pointer;
        }

        .rdp-button.rdp-day:not(.rdp-day_outside):not(.rdp-day_range_middle) {
          background-color: ${theme.colors.offWhite};
          box-shadow: 0px 1px 1px rgba(0, 14, 51, 0.05);
          &.rdp-day_selected {
            background-color: ${theme.colors.primary};
            color: ${theme.colors.offWhite};
          }
        }

        .rdp-day_outside.rdp-day_selected.rdp-day {
          opacity: 0.5;
        }

        .rdp-day_selected.rdp-day.rdp-day_range_middle {
          color: ${theme.colors.primary};
          background-color: ${theme.colors.bgPrimary};
          border-radius: 0;
        }

        .rdp-day_outside {
          color: ${theme.colors.placeholder};
        }
      }
    }
  `}
`
