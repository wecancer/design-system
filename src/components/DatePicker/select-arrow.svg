<?xml version="1.0" encoding="UTF-8"?>
<svg width="7" height="7" viewBox="0 0 7 7" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path
    d="M5.71967 0.280329L0.28033 5.71967C-0.192143 6.19214 0.142482 7 0.81066 7H5.5C6.32843 7 7 6.32843 7 5.5V0.81066C7 0.142482 6.19214 -0.192145 5.71967 0.280329Z"
    fill="#0080EB"
  />
</svg>