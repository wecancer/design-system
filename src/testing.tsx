import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
