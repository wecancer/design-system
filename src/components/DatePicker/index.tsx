import React, {useState} from 'react'
import * as ReactDayPicker from 'react-day-picker'
import styled from 'styled-components'
import * as fns from 'date-fns'
import ptLocale from 'date-fns/locale/pt-BR'

import Input from '../Form/Input/Input'
import DatePickerStyledContainer from './DatePickerStyledContainer'

const defaultDateFormat = 'dd/MM/yyyy'

const Container = styled.div`
  position: relative;
`

export type DatePickerValue = Date

type ModifierStatus = ReactDayPicker.ActiveModifiers

type OnChange = {
  value?: DatePickerValue
}

export type Props = {
  label?: string
  value?: DatePickerValue
  onChange?(params: OnChange): void
  fromYear?: number
  toYear?: number
}

const DatePicker = ({label, value, onChange, fromYear, toYear}: Props) => {
  if (value && !(value instanceof Date)) {
    throw new Error('The value attribute should be a date instance.')
  }

  const [isOpen, setIsOpen] = useState<boolean>(false)
  const {inputProps, dayPickerProps} = ReactDayPicker.useInput({
    format: 'dd/MM/yyyy',
    defaultSelected: value || undefined,
    required: true,
  })

  const handleChange = (data?: DatePickerValue): void => {
    if (typeof onChange === 'function') {
      onChange({value: data})
    }
  }

  return (
    <Container
      role="button"
      tabIndex={0}
      onClick={() => setIsOpen(true)}
      onKeyDown={() => null}
      onBlur={({currentTarget, relatedTarget}) => {
        if (currentTarget.contains(relatedTarget as Node)) return
        setIsOpen(false)
      }}
    >
      <Input
        type="text"
        {...inputProps}
        label={label}
        value={`${inputProps.value}`}
        iconButtonRight={{
          onClick(e) {
            e.stopPropagation()
            setIsOpen(!isOpen)
          },
          type: 'calendar',
        }}
        onChange={({event}) => inputProps.onChange?.(event as React.ChangeEvent<HTMLInputElement>)}
        onBlur={(e) => {
          inputProps.onBlur?.(e as React.FocusEvent<HTMLInputElement>)
          const val = inputProps?.value || undefined
          if (typeof val === 'string' && fns.isMatch(val, defaultDateFormat)) {
            const [day, month, year] = val.split('/')
            handleChange(new Date(+year, +month - 1, +day))
            return
          }
          handleChange(undefined)
        }}
      />
      {isOpen && (
        <DatePickerStyledContainer>
          <ReactDayPicker.DayPicker
            {...dayPickerProps}
            mode="single"
            toYear={toYear}
            showOutsideDays
            selected={value}
            locale={ptLocale}
            fromYear={fromYear}
            captionLayout="dropdown"
            onSelect={(date?: Date, selectedDay?: Date, modifiers?: ModifierStatus, evt?: React.MouseEvent) => {
              if (date) {
                evt?.stopPropagation()
                setIsOpen(false)
                handleChange(date)
              }
            }}
          />
        </DatePickerStyledContainer>
      )}
    </Container>
  )
}

export default DatePicker
