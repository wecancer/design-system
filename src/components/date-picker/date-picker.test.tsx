import React from 'react'
import DatePicker from '.'
import RangeDatePicker from './range'
import { render, fireEvent } from '../../testing'

describe('<DatePicker />', () => {
  it('should dispatch an expection when the value has the invalid type', () => {
    const spy = jest.spyOn(console, 'error')
    spy.mockImplementation(() => {})
    try {
      // eslint-disable-next-line
      // @ts-ignore: Unreachable code error
      expect(() => render(<DatePicker value="10/10/2010" />)).toThrowError(
        'The value attribute should be a date instance.',
      )
    } catch (err) {
      // do nothing
    }
    spy.mockRestore()
  })

  it('should open the calendar when the input has been focused', () => {
    const { container } = render(<DatePicker />)
    const inputTexts = container.querySelectorAll('input[type="text"]')
    expect(inputTexts).toHaveLength(1)

    expect(container.querySelector('.rdp')).toBeNull()
    fireEvent.click(inputTexts[0] as HTMLElement)
    expect(container.querySelector('.rdp')).toBeInstanceOf(HTMLElement)
  })

  it('should dispatch an event when the onChange is triggered', () => {
    const callback = jest.fn()
    const { container } = render(
      <DatePicker value={new Date()} onChange={callback} />,
    )
    const input = container.querySelector('input[type="text"]') as HTMLElement

    fireEvent.click(input)
    fireEvent.change(input, { target: { value: '10/10/2010' } })
    fireEvent.blur(input)
    expect(callback).toHaveBeenCalledTimes(1)
  })

  it('should open the calendar when click on the calendar icon button', () => {
    const callback = jest.fn()
    const { container } = render(
      <DatePicker value={new Date()} onChange={callback} />,
    )
    const button = container.querySelector('button') as HTMLElement

    expect(container.querySelector('.rdp')).toBeNull()
    fireEvent.click(button as HTMLElement)
    expect(container.querySelector('.rdp')).toBeInstanceOf(HTMLElement)
  })

  it('should expect the new value when change the calendar', () => {
    const { container } = render(
      <DatePicker
        value={new Date()}
        onChange={({ value }) => expect(value).toBeInstanceOf(Date)}
      />,
    )

    const input = container.querySelector('input[type="text"]') as HTMLElement
    fireEvent.click(input)
    const button = container.querySelector('.rdp-cell button') as HTMLElement
    fireEvent.click(button)
  })

  it('should set the value undefined when the user type an invalid date format', () => {
    const { container } = render(
      <DatePicker
        value={new Date()}
        onChange={({ value }) => expect(value).toBeUndefined()}
      />,
    )
    const input = container.querySelector('input[type="text"]') as HTMLElement

    fireEvent.click(input)
    fireEvent.change(input, { target: { value: '50/30/9999' } })
    fireEvent.blur(input)
  })

  it('should defines the input label to datepicker component', () => {
    const { getByText } = render(
      <DatePicker
        label="Birthday"
        value={new Date()}
        onChange={({ value }) => expect(value).toBeUndefined()}
      />,
    )
    const label = getByText('Birthday')

    expect(label).toBeInstanceOf(HTMLDivElement)
  })
})

describe('<RangeDatePicker />', () => {
  it('should open the calendar when the input has been focused', () => {
    const { container } = render(
      <RangeDatePicker value={{ from: undefined, to: undefined }} />,
    )
    const inputTexts = container.querySelectorAll('input[type="text"]')
    expect(inputTexts).toHaveLength(2)

    expect(container.querySelector('.rdp')).toBeNull()
    fireEvent.click(inputTexts[0] as HTMLElement)
    expect(container.querySelector('.rdp')).toBeInstanceOf(HTMLElement)
  })

  it('should defines the input label to rander datepicker component', () => {
    const { getByText } = render(
      <RangeDatePicker
        endLabel="End"
        startLabel="Start"
        value={{ from: undefined, to: undefined }}
        onChange={({ value }) => expect(value).toBeUndefined()}
      />,
    )
    const end = getByText('End')
    const start = getByText('Start')

    expect(end).toBeInstanceOf(HTMLDivElement)
    expect(start).toBeInstanceOf(HTMLDivElement)
  })
})

describe('Snapshot <DatePicker />', () => {
  it('should take a snapshot', () => {
    const { container } = render(
      <>
        <DatePicker fromYear={2022} toYear={2025} />
      </>,
    )
    const input = container.querySelector('input[type="text"]') as HTMLElement
    fireEvent.change(input, { target: { placeholder: '22/02/2022' } })
    fireEvent.click(input)
    expect(container).toMatchSnapshot()
  })
})
