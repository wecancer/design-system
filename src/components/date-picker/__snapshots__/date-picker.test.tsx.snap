// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot <DatePicker /> should take a snapshot 1`] = `
<div>
  <div
    class="sc-iCfMLu FBMMr"
    role="button"
    tabindex="0"
  >
    <div
      class="sc-gsDKAQ kdpXcC"
    >
      <input
        class="sc-dkPtRN kHFFrW"
        placeholder="22/02/2022"
        type="text"
        value=""
      />
      <button
        class="sc-eCImPb jWvkiQ"
        type="button"
      >
        <div
          class="sc-bdvvtL bYGNWJ icon-right"
          data-icon-name="calendar"
        >
          <div>
            mock svg
          </div>
        </div>
      </button>
    </div>
    <div
      class="sc-gKclnd cyCAXb"
    >
      <div
        class="rdp"
      >
        <div
          class="rdp-months"
        >
          <div
            class="rdp-month rdp-caption_start rdp-caption_end"
          >
            <div
              class="rdp-caption"
            >
              <div
                class="rdp-caption_dropdowns"
              >
                <div
                  class="rdp-vhidden"
                >
                  <h2
                    aria-atomic="true"
                    aria-live="polite"
                    class="rdp-caption_label"
                    id="11"
                  >
                    março 2022
                  </h2>
                </div>
                <div
                  class="rdp-dropdown_month"
                >
                  <span
                    class="rdp-vhidden"
                  >
                    Month: 
                  </span>
                  <select
                    aria-label="Month: "
                    class="rdp-dropdown"
                  >
                    <option
                      value="0"
                    >
                      janeiro
                    </option>
                    <option
                      value="1"
                    >
                      fevereiro
                    </option>
                    <option
                      value="2"
                    >
                      março
                    </option>
                    <option
                      value="3"
                    >
                      abril
                    </option>
                    <option
                      value="4"
                    >
                      maio
                    </option>
                    <option
                      value="5"
                    >
                      junho
                    </option>
                    <option
                      value="6"
                    >
                      julho
                    </option>
                    <option
                      value="7"
                    >
                      agosto
                    </option>
                    <option
                      value="8"
                    >
                      setembro
                    </option>
                    <option
                      value="9"
                    >
                      outubro
                    </option>
                    <option
                      value="10"
                    >
                      novembro
                    </option>
                    <option
                      value="11"
                    >
                      dezembro
                    </option>
                  </select>
                  <div
                    aria-hidden="true"
                    class="rdp-caption_label"
                  >
                    março
                    <svg
                      class="rdp-dropdown_icon"
                      data-testid="iconDropdown"
                      height="8px"
                      viewBox="0 0 120 120"
                      width="8px"
                    >
                      <path
                        d="M4.22182541,48.2218254 C8.44222828,44.0014225 15.2388494,43.9273804 19.5496459,47.9996989 L19.7781746,48.2218254 L60,88.443 L100.221825,48.2218254 C104.442228,44.0014225 111.238849,43.9273804 115.549646,47.9996989 L115.778175,48.2218254 C119.998577,52.4422283 120.07262,59.2388494 116.000301,63.5496459 L115.778175,63.7781746 L67.7781746,111.778175 C63.5577717,115.998577 56.7611506,116.07262 52.4503541,112.000301 L52.2218254,111.778175 L4.22182541,63.7781746 C-0.0739418023,59.4824074 -0.0739418023,52.5175926 4.22182541,48.2218254 Z"
                        fill="currentColor"
                        fill-rule="nonzero"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="rdp-dropdown_month"
                >
                  <span
                    class="rdp-vhidden"
                  >
                    Year: 
                  </span>
                  <select
                    aria-label="Year: "
                    class="rdp-dropdown"
                  >
                    <option
                      value="2022"
                    >
                      2022
                    </option>
                    <option
                      value="2023"
                    >
                      2023
                    </option>
                    <option
                      value="2024"
                    >
                      2024
                    </option>
                    <option
                      value="2025"
                    >
                      2025
                    </option>
                  </select>
                  <div
                    aria-hidden="true"
                    class="rdp-caption_label"
                  >
                    2022
                    <svg
                      class="rdp-dropdown_icon"
                      data-testid="iconDropdown"
                      height="8px"
                      viewBox="0 0 120 120"
                      width="8px"
                    >
                      <path
                        d="M4.22182541,48.2218254 C8.44222828,44.0014225 15.2388494,43.9273804 19.5496459,47.9996989 L19.7781746,48.2218254 L60,88.443 L100.221825,48.2218254 C104.442228,44.0014225 111.238849,43.9273804 115.549646,47.9996989 L115.778175,48.2218254 C119.998577,52.4422283 120.07262,59.2388494 116.000301,63.5496459 L115.778175,63.7781746 L67.7781746,111.778175 C63.5577717,115.998577 56.7611506,116.07262 52.4503541,112.000301 L52.2218254,111.778175 L4.22182541,63.7781746 C-0.0739418023,59.4824074 -0.0739418023,52.5175926 4.22182541,48.2218254 Z"
                        fill="currentColor"
                        fill-rule="nonzero"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <table
              aria-labelledby="11"
              class="rdp-table"
              role="grid"
            >
              <thead
                class="rdp-head"
              >
                <tr
                  class="rdp-head_row"
                >
                  <th
                    class="rdp-head_cell"
                    scope="col"
                  >
                    <span
                      aria-hidden="true"
                    >
                      dom
                    </span>
                    <span
                      class="rdp-vhidden"
                    >
                      domingo
                    </span>
                  </th>
                  <th
                    class="rdp-head_cell"
                    scope="col"
                  >
                    <span
                      aria-hidden="true"
                    >
                      seg
                    </span>
                    <span
                      class="rdp-vhidden"
                    >
                      segunda-feira
                    </span>
                  </th>
                  <th
                    class="rdp-head_cell"
                    scope="col"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ter
                    </span>
                    <span
                      class="rdp-vhidden"
                    >
                      terça-feira
                    </span>
                  </th>
                  <th
                    class="rdp-head_cell"
                    scope="col"
                  >
                    <span
                      aria-hidden="true"
                    >
                      qua
                    </span>
                    <span
                      class="rdp-vhidden"
                    >
                      quarta-feira
                    </span>
                  </th>
                  <th
                    class="rdp-head_cell"
                    scope="col"
                  >
                    <span
                      aria-hidden="true"
                    >
                      qui
                    </span>
                    <span
                      class="rdp-vhidden"
                    >
                      quinta-feira
                    </span>
                  </th>
                  <th
                    class="rdp-head_cell"
                    scope="col"
                  >
                    <span
                      aria-hidden="true"
                    >
                      sex
                    </span>
                    <span
                      class="rdp-vhidden"
                    >
                      sexta-feira
                    </span>
                  </th>
                  <th
                    class="rdp-head_cell"
                    scope="col"
                  >
                    <span
                      aria-hidden="true"
                    >
                      sab
                    </span>
                    <span
                      class="rdp-vhidden"
                    >
                      sábado
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                class="rdp-tbody"
              >
                <tr
                  class="rdp-row"
                >
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day rdp-day_outside"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        27
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        27º fevereiro (domingo)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day rdp-day_outside"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        28
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        28º fevereiro (segunda-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        1
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        1º março (terça-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        2
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        2º março (quarta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        3
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        3º março (quinta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        4
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        4º março (sexta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        5
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        5º março (sábado)
                      </span>
                    </button>
                  </td>
                </tr>
                <tr
                  class="rdp-row"
                >
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        6
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        6º março (domingo)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        7
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        7º março (segunda-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        8
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        8º março (terça-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        9
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        9º março (quarta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        10
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        10º março (quinta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        11
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        11º março (sexta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        12
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        12º março (sábado)
                      </span>
                    </button>
                  </td>
                </tr>
                <tr
                  class="rdp-row"
                >
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        13
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        13º março (domingo)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        14
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        14º março (segunda-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        15
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        15º março (terça-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        16
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        16º março (quarta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        17
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        17º março (quinta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        18
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        18º março (sexta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        19
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        19º março (sábado)
                      </span>
                    </button>
                  </td>
                </tr>
                <tr
                  class="rdp-row"
                >
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        20
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        20º março (domingo)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        21
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        21º março (segunda-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        22
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        22º março (terça-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        23
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        23º março (quarta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        24
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        24º março (quinta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day rdp-day_today"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        25
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        25º março (sexta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        26
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        26º março (sábado)
                      </span>
                    </button>
                  </td>
                </tr>
                <tr
                  class="rdp-row"
                >
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        27
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        27º março (domingo)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        28
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        28º março (segunda-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        29
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        29º março (terça-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        30
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        30º março (quarta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        31
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        31º março (quinta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day rdp-day_outside"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        1
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        1º abril (sexta-feira)
                      </span>
                    </button>
                  </td>
                  <td
                    class="rdp-cell"
                  >
                    <button
                      class="rdp-button_reset rdp-button rdp-day rdp-day_outside"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        2
                      </span>
                      <span
                        class="rdp-vhidden"
                      >
                        2º abril (sábado)
                      </span>
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
